Find a functionally equivalent code:com.android.internal.widget.WaveView.vibrate:COMMENT
Method Modifier: private     internal    
Comment:/**
 * Triggers haptic feedback.
 */

@@ -4,9 +4,9 @@
         if (mVibrator == null) {
             mVibrator = (android.os.Vibrator) getContext().getSystemService(Context.VIBRATOR_SERVICE);
         }
-        mVibrator.vibrate(duration);
+        mVibrator.vibrate(duration, VIBRATION_ATTRIBUTES);
     }
 }

